src/components/inspectors/TimerExpression.vue

```diff
@@ -295,9 +307,9 @@ export default {
         (this.ends === 'ondate' ? this.endDate : '')
       );
     },
-    getWeekDayDate(date, isoWeekDay) {
+    getWeekDayDate(startDate, isoWeekDay) {
```

Comment:
Problem seems to be with this function; it appears it's adding a day to each selected weekday.